lib/spack/spack/build_systems/intel.py

```diff
@@ -103,6 +136,742 @@ def _filtered_components(self):
+                matches = re.search(
+                    r'gcc.* ([0-9]+\.[0-9]+\.[0-9]+).*',
+                    gcc('--version', output=str))
+                if matches:
+                    gcc_version = Version(matches.groups()[0])
+                    if gcc_version >= Version('4.7'):
+                        tbb_abi = 'gcc4.7'
+                    elif gcc_version >= Version('4.4'):
+                        tbb_abi = 'gcc4.4'
+                else:
+                    tbb_abi = 'gcc4.1'     # unlikely, one hopes.
+
+                # Alrighty then ...
+                d = Prefix(join_path(d, tbb_abi))
+
+        if relative:
+            d = self.relative_path(d)
+
+        debug_print(d)
+        return d
+
+    @property
+    def component_include_dir(self, component=None, relative=False):
+        d = self.component_dir(component)
+
+        if component == 'mpi' or 'intel-mpi' in self.name:
+            d = d.intel64
+        d = d.include
+
+        if relative:
+            d = self.relative_path(d)
+
+        debug_print(d)
+        return d
+
+    @property
+    def openmp_libs(self):
+        '''Supply LibraryList for linking OpenMP'''
+
+        if '%intel' in self.spec:
+            omp_libnames = ['libiomp5']
+            # NB: Hunting down explicit library files may be the Spack-way of
+            # doing things, but be aware that "{icc|ifort} --help openmp"
+            # steers us towards options instead: -qopenmp-link={dynamic,static}
+            #
+            # Note about search root: For MKL, the directory
+            # "$MKLROOT/../compiler" will be present even for an MKL-only
+            # product installation (as opposed to one being ghosted via
+            # packages.yaml), specificially to provide the 'iomp5' libs.
+
+            omp_libs = find_libraries(
+                omp_libnames,
+                root=self.component_lib_dir(component='compiler'),
+                shared=self._want_shared)
+
+        elif '%gcc' in self.spec:
+            gcc = Executable(self.compiler.cc)
+            omp_lib_path = gcc(
+                '--print-file-name', 'libgomp.%s' % dso_suffix, output=str)
+            omp_libs = LibraryList(omp_lib_path)
+
+        if len(omp_libs) < 1:
+            raise_lib_error('Cannot locate OpenMP libraries:', omp_libnames)
+
+        debug_print(omp_libs)
+        return omp_libs
+
+    @property
+    def tbb_libs(self):
+        '''Supply LibraryList for linking TBB'''
+
+        tbb_lib = find_libraries(
+            ['libtbb'], root=self.component_lib_dir(component='tbb'))
+        # NB: shared=False is not and has never been supported for TBB:
+        # https://www.threadingbuildingblocks.org/faq/there-version-tbb-provides-statically-linked-libraries
+        #
+        # NB2: Like icc with -qopenmp, so does icpc steer us towards using an
+        # option: "icpc -tbb"
+
+        gcc = Executable('gcc')     # must be gcc, not self.compiler.cc
```

Comment:
what if we work with `clang` on macOS? If the combination is unsupported, it should be added as a conflict in `intel-mkl` and `intel-parallel-studio`.